lib/events/filesessions/filestream.go

```diff
@@ -119,12 +119,12 @@ func (h *Handler) CompleteUpload(ctx context.Context, upload events.StreamUpload
 		return trace.Wrap(err)
 	}
 	defer func() {
-		if err := f.Close(); err != nil {
-			h.WithError(err).Errorf("Failed to close file %q.", uploadPath)
-		}
 		if err := utils.FSUnlock(f); err != nil {
 			h.WithError(err).Errorf("Failed to unlock filesystem lock.")
 		}
+		if err := f.Close(); err != nil {
+			h.WithError(err).Errorf("Failed to close file %q.", uploadPath)
+		}
```

Comment:
Is this change related to the PR?

Reply:
Not directly, I just noticed these "failed to unlock filesystem lock" errors in the logs when working on this and tracked it down to these re-ordered calls. These errors surfaced after the security changes a few weeks back - previously they just weren't logged. Felt not worth it to make a separate PR for this so I included the fix here, I can remove though if you feel strongly otherwise.